### ISSUE TITLE
Allow to use existing videos as a source of daily records.

### ISSUE THREAD

**Issue by brenocrs** [KyleKun/one_second_diary#8]
Description:
Allow videos that are already in my gallery to be used as a source for the daily records.

Reason:
It's already very difficult to capture a special moment, 1 second so even more, usually they happen in a bigger moment, with bigger videos, if it was possible to take a bigger video already existing and "cut" it in the exact second, the daily records would be with fewer gaps.

Functionality:
- [ ] Allow to use existing videos as a source of daily records.
- [ ] From these existing videos, it will be possible to edit and cut the exact moment.

**Comment by KyleKun** [comment]
Hey @brenocrs !

Thanks for using the app and proposing this feature.

I already had in mind doing that, selecting the video and cropping it to a second should not be a hard task, the problem is with ffmpeg.

The way the videos are compiled together is through a lossless command, and it fails if there's a video with different resolution. 

Other lossless commands are way too slow than the one used in the app. 

Basically, that's the reason why I've also locked the camera orientation. 

Anyways, I'll try to find an alternative solution to this, or even process the uploaded video metadata to avoid such problems.

If you want to help or happen to know a solution please don't hesitate contributing! 😁

**Comment by brenocrs** [comment]
@KyleKun ,what about:

1.  Identify the video orientation: using [ffprobe or exif-metadata](https://stackoverflow.com/questions/5287603/how-to-extract-orientation-information-from-videos)
2.  If is not in landscape, use the [transpose](https://stackoverflow.com/questions/3937387/rotating-videos-with-ffmpeg) or [rotation](https://gist.github.com/ViktorNova/1dd68a2ec99781fd9adca49507c73ee2) function in the ffmpeg

**Comment by KyleKun** [comment]
> @KyleKun ,what about:
> 
> 1.  Identify the video orientation: using [ffprobe or exif-metadata](https://stackoverflow.com/questions/5287603/how-to-extract-orientation-information-from-videos)
> 2.  If is not in landscape, use the [transpose](https://stackoverflow.com/questions/3937387/rotating-videos-with-ffmpeg) or [rotation](https://gist.github.com/ViktorNova/1dd68a2ec99781fd9adca49507c73ee2) function in the ffmpeg
> 

Thanks for the suggestion, I'll take a look into it for the next update!
